### ISSUE TITLE
Uncrustify Issue Reporter Bot

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1810]
The idea is to have a GitHub app that lets users report formatting bugs using a webapp instead of creating issues here. The app should thereafter create the issues here with a template (similar to what we had used for issues automatically created from our side #1223) or how @CDanU used to write them with: `I:... O:... E:... C:...` blocks (i.e. Input, Output, Expected, Config) (please correct me here).

Anyway, the app should:
- require people all this input fields before creating the issue
- validate that's a real issue (if the issue is fixed in the latest master)
- reduce the user's config
- bisect the issue if the issue is failing now but it was working in an older version
- if the issue is reproducible, the app creates a PR with the incoming test files placed in the right folders and in a `staging.test` testing suite which is ran by the CI builds but here we expect all tests to fail (i.e. tests from the staging will fail if they pass - condition is reversed)
- we fix the issue and the CI will fail with that test passing
- we move the test to a dedicated folder language or to another testing suite `incoming.test` or `regression.test` (what we have in our fork).
- when the test is fixed, the issue will be closed by the app

With this new pipeline:
- increase our testing coverage
- top people from creating non-issues
- help ourselves improving our workflow
- focus more on fixing issues and implementing new features rather than doing a lot of help desk work

What do you guys think?

PS: I remembered about this when I saw a message from @guy-maurel closing an unfinished conversation we had here #1463 that I forgot about. Sorry Daniel about that. You should create a new separate issue with your idea and I'll chip in there.

**Comment by gmaurel** [comment]
It would be fine.

**Comment by CDanU** [comment]
Yes I'd like to have that as well, so far I haven't found the server for that. 
I have to admit that I didn't look that much, there are apparently some services that provide a nodejs server (netlify, zeit.co). The zeit.co also provides a docker service but I don't know how much access we actually get there, those seem to be more geared towards web deployment.

For the ideas that I had I needed to provide those services a key so that they could alter a submitted PRs, I am skeptical about doing that.
